std/array.d

```diff
@@ -3008,11 +3008,17 @@ if (isDynamicArray!A)
         this.arr = arr;
     }
 
-    auto opDispatch(string fn, Args...)(Args args) if (is(typeof(mixin("impl." ~ fn ~ "(args)"))))
+    template opDispatch(string fn)
     {
-        // we do it this way because we can't cache a void return
-        scope(exit) *this.arr = impl.data;
-        mixin("return impl." ~ fn ~ "(args);");
+        // This is a different name plus an alias to work around issue 14604.
+        auto opDispatchImpl(Args...)(Args args)
```

Comment:
I'd think the latter. However there were discussions about `introducesMember` vs `hasMember` so it may be by design of `hasMember`.